AppServer/google/appengine/ext/datastore_admin/backup_handler.py

```diff
@@ -129,6 +130,19 @@
 BACKUP_RESTORE_HANDLER = __name__ + '.RestoreEntity.map'
 
 
+def get_service_account_names():
+  """ AppScale: Fetch list of service accounts from IAM API. """
+  project_id = app_identity.get_application_id()
+  iam_location = 'https://127.0.0.1:17441'
+  url = iam_location + '/v1/projects/{}/serviceAccounts'.format(project_id)
+  token = app_identity.get_access_token(
+      ['https://www.googleapis.com/auth/cloud-platform'])[0]
+  headers = {'Authorization': 'Bearer {}'.format(token)}
+  response = urlfetch.fetch(url, headers=headers, validate_certificate=False)
+  accounts = json.loads(response.content)['accounts']
```

Comment:
not sure how pedantic you want to be, but this could throw an exception if 'accounts' isn't in the json